nucleus/common/common-util/src/main/java/com/sun/enterprise/security/store/AsadminSecurityUtil.java

```diff
@@ -252,7 +253,34 @@ private void init(final char[] commandLineMasterPassword, final boolean isPrompt
          * The keystore has been opened successfully, using passwordToUse.
          * Open the truststore with that password.
          */
-        asadminTruststore = openTruststore(passwordToUse);
+        try {
+            asadminTruststore = openTruststore(passwordToUse);
+        } catch (IOException e) {
+            if (System.getProperty(SystemPropertyConstants.CLIENT_TRUSTSTORE_PROPERTY) == null) {
+                logger.log(Level.WARNING, String.format("Error when reading truststore, exception:%s. Now recreating file", e));
+                recreateDefaultTrustStore(passwordToUse);
+            }
```

Comment:
Suggestion: Add explanative comment and don't eat the exception if no remedial action being taken
```suggestion
            // If we're using the default trust store try to recreate it, otherwise just throw the error
            if (System.getProperty(SystemPropertyConstants.CLIENT_TRUSTSTORE_PROPERTY) == null) {
                logger.log(Level.WARNING, String.format("Error when reading truststore, exception:%s. Now recreating file", e));
                recreateDefaultTrustStore(passwordToUse);
            } else {
                throw e;
            }
```